### PR TITLE
Adds dynamic storage prorvisioner and cloud provider capability

### PR DESCRIPTION
**Pull request by puja108**
This:
- adds default storage class admission control plugin
- adds cloudprovider flag
- applies default storage class if available

towards https://github.com/giantswarm/giantswarm/issues/1588

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -784,7 +797,7 @@ write_files:
               /usr/bin/docker run --net=host --rm -v /srv:/srv $KUBECTL apply -f /srv/$manifest
               [ "$?" -ne "0" ]
           do
-              echo "failed to apply /src/$manifest, retrying in 5 sec"
+              echo "failed to apply /srv/$manifest, retrying in 5 sec"
```
Comment (puja108) [inline templates.go]: this is a minor boyscouting fix

File: templates.go [hunk 2]
```diff
@@ -1107,6 +1120,7 @@ coreos:
       --logtostderr=true \
       --machine-id-file=/rootfs/etc/machine-id \
       --cadvisor-port=4194 \
+      --cloud-provider={{.Cluster.Kubernetes.CloudProvider}} \
```
Comment (rossf7) [inline templates.go]: This won't work. The templating is for the Kubernetes block in clustertpr and it doesn't have a CloudProvider.

Comment (puja108) [inline templates.go]: should be solved by https://github.com/giantswarm/clustertpr/pull/28 ?

Comment (rossf7) [inline templates.go]: Yes, that will fix it.

Comment (puja108) [inline templates.go]: done

**Comment by rossf7** [comment]
FYI @nhlfr since you're working on the move to Ignition :)

**Pull request merged**